src/logic/calculate.js

```diff
@@ -0,0 +1,16 @@
+import operate from './operate';
+
+const calculate = ({ total, next, operation }, buttonName) => {
+  if (buttonName === '+/-') {
+    const negTotal = (total) * -1;
+    const negNext = (next) * -1;
+    return { total: negTotal, next: negNext };
+  }
+  if (buttonName === '%' || buttonName === '÷' || buttonName === 'X' || buttonName === '-' || buttonName === '+') {
+    const totalResult = operate(total, next, operation);
+    return { total: totalResult.toString() };
+  }
```

Comment:
### Heads-Up:
**we need to add more cases;** 
1. if the `buttonName` = `A/C`, we should return `0`
2. if the `buttonName`  = `.` we should return `0.`